tests/test_utils.py

```diff
@@ -226,3 +228,30 @@ def forward(self, x):
                 save(model.state_dict(), save_path, safe_serialization=True)
                 self.assertEqual(len(log.records), 1)
                 self.assertIn("Removed shared tensor", log.output[0])
+
+    @require_cuda
+    def test_offload_save_state_dict(self):
+        device_map = {
```

Comment:
You don't need to add this test. You can simply change the following test https://github.com/huggingface/accelerate/blob/cd515812483aedd070933c3367366a0b5ef43daa/tests/test_accelerator.py#L160 by removing the assertRaises since this PR will enable saving offloaded model !

Reply:
Must have missed this test, that works very well. The state dict test is removed and the assertRaises has been removed too.